src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1060,48 +1186,26 @@ private static string GetInputTypeString(InputType inputType)
         }
 
         private static IEnumerable<SelectListItem> UpdateSelectListItemsWithDefaultValue(
+            ModelExplorer modelExplorer,
             IEnumerable<SelectListItem> selectList,
-            object defaultValue,
-            bool allowMultiple,
-            out ICollection<string> selectedValues)
+            IReadOnlyCollection<string> currentValues)
         {
-            IEnumerable defaultValues;
-            if (allowMultiple)
-            {
-                defaultValues = defaultValue as IEnumerable;
-                if (defaultValues == null || defaultValues is string)
-                {
-                    throw new InvalidOperationException(
-                        Resources.FormatHtmlHelper_SelectExpressionNotEnumerable("expression"));
-                }
-            }
-            else
-            {
-                defaultValues = new[] { defaultValue };
-            }
-
-            var values =
-                defaultValues.OfType<object>().Select(value => Convert.ToString(value, CultureInfo.CurrentCulture));
-
-            // ToString() by default returns an enum value's name.  But selectList may use numeric values.
-            var enumValues = defaultValues.OfType<Enum>().Select(value => value.ToString());
-            values = values.Concat(enumValues);
-
-            selectedValues = new HashSet<string>(values, StringComparer.OrdinalIgnoreCase);
-
             // Perform deep copy of selectList to avoid changing user's Selected property values.
             var newSelectList = new List<SelectListItem>();
             foreach (SelectListItem item in selectList)
             {
-                var newItem = new SelectListItem
+                var value = item.Value ?? item.Text;
+                var selected = currentValues.Contains(value);
```

Comment:
the collection comes from `GetCurrentValues()` and that provides a `HashSet` that ignores case.  it is possible to be explicit here (and in `OptionTagHelper`) using the Linq extension method.  but it isn't clear to me why we'd override a developer who _decides_ to provide a case-sensitive collection.